ffmpeg_tools/validation.py

```diff
@@ -294,9 +310,22 @@ def validate_resolution(src_resolution, target_resolution):
     Validate if aspect ratio of source resolution and
     target resolution are the same.
     """
-    if formats.get_effective_aspect_ratio(src_resolution) == \
-            formats.get_effective_aspect_ratio(target_resolution):
+    if (
+        src_resolution is None
+        or target_resolution is None
```

Comment:
Validation shouldn't fail in case of target resolution. It should assume, that parameter doesn't change, if not specified. Related to review of https://github.com/golemfactory/golem/pull/4954#pullrequestreview-326881496

Reply:
It was already failing before this commit. The only thing I changed here is that it raises a validation error (`InvalidResolution`) rather than a random exception from some function down the call stack that actually tries to use the value and fails. The only reason I changed it was that I was getting these random exceptions when running tests in Golem and I thought that this is something that should be stopped by a validation instead of just letting it crash.

And as for https://github.com/golemfactory/golem/pull/4954 - it was still implemented with an assumption that resolution is mandatory and this still holds here. If done the way you propose - yeah, I'd have to handle missing resolution. But even if #4954 is going to be reimplemented with different assumptions, I'd do it on top of other pull requests rather than as a review fix and in that case it makes sense for this validation to stay in the code until the assumptions about video codec and resolution are changed.